master/internal/db/postgres_experiments_intg_test.go

```diff
@@ -6,16 +6,326 @@ package db
 	"github.com/determined-ai/determined/proto/pkg/trialv1"
 )
 
+func TestGetExperiments(t *testing.T) {
+	etc.SetRootPath(rootFromDB)
+	db := MustResolveTestPostgres(t)
+	MustMigrateTestPostgres(t, db, migrationsFromDB)
+
+	// Add a mock user.
+	user := requireMockUser(t, db)
+	const (
+		activeState = "ACTIVE"
+		unarchived  = "true"
+		adesc       = "a description"
+		alabel      = "a label"
+		aname       = "a name"
+		orderByID   = "id ASC"
+	)
+
+	tests := []struct {
+		name           string
+		n              int
+		exps           []model.Experiment
+		require        func(*testing.T, *experimentv1.Experiment)
+		stateFilter    string
+		archivedFilter string
+		labelFilter    string
+		descFilter     string
+		nameFilter     string
+		offset, limit  int
+	}{
+		{
+			name: "empty result",
+			n:    0,
+			exps: func() []model.Experiment { return nil }(),
+		},
+		{
+			name: "unfiltered result",
+			n:    3,
+			exps: func() []model.Experiment {
+				var exps []model.Experiment
+				for i := 0; i < 3; i++ {
+					exps = append(exps, model.Experiment{
+						JobID:                model.NewJobID(),
+						State:                model.ActiveState,
+						Config:               mockExpconf(),
+						ModelDefinitionBytes: []byte{1, 0, 1, 0, 1, 0},
+						StartTime:            time.Now().Add(-time.Hour),
+						OwnerID:              &user.ID,
+						Username:             user.Username,
+					})
+				}
+				return exps
+			}(),
+		},
+		{
+			name: "fitler by state",
+			n:    3,
+			exps: func() []model.Experiment {
+				var exps []model.Experiment
+				for i := 0; i < 3; i++ {
+					exps = append(exps, model.Experiment{
+						JobID:                model.NewJobID(),
+						State:                model.CanceledState,
+						Config:               mockExpconf(),
+						ModelDefinitionBytes: []byte{1, 0, 1, 0, 1, 0},
+						StartTime:            time.Now().Add(-time.Hour),
+						OwnerID:              &user.ID,
+						Username:             user.Username,
+					})
+				}
+				return exps
+			}(),
+			require: func(t *testing.T, e *experimentv1.Experiment) {
+				require.Equal(t, experimentv1.State_STATE_CANCELED, e.State)
+			},
+			stateFilter: string(model.CanceledState),
+		},
+		{
+			name: "fitler by archived",
+			n:    3,
+			exps: func() []model.Experiment {
+				var exps []model.Experiment
+				for i := 0; i < 3; i++ {
+					exps = append(exps, model.Experiment{
+						JobID:                model.NewJobID(),
+						State:                model.ActiveState,
+						Config:               mockExpconf(),
+						ModelDefinitionBytes: []byte{1, 0, 1, 0, 1, 0},
+						StartTime:            time.Now().Add(-time.Hour),
+						OwnerID:              &user.ID,
+						Username:             user.Username,
+						Archived:             true,
+					})
+				}
+				return exps
+			}(),
+			require: func(t *testing.T, e *experimentv1.Experiment) {
+				require.Equal(t, true, e.Archived)
+			},
+			archivedFilter: "true",
+		},
+		{
+			name: "fitler by labels",
```

Comment:
:facepalm: as is tradition, i have horrible misspellings with @dzhu as a reviewer.